src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,488 @@
+        """
+        if 'GIAO' in self._keywords():
+            return 3
+        elif 'CSGT' in self._keywords():
+            return 6
+        return 0
+
+    def _nuclear_names(self):
+        """Names of nuclei present in the molecule.
+
+        O1
+        H2
+        H3
+        """
+        return [self.pt.element[Z]+str(i) for i, Z in
+                enumerate(self.ccdata.atomnos, start=1)]
+
+    def _atomic_nos(self):
+        """Atomic numbers of elements."""
+        return [str(Z) for Z in self.ccdata.atomnos]
+
+    def _nuclear_charges(self):
+        """Nuclear charges."""
+        if hasattr(self.ccdata, 'coreelectrons'):
+            return ['{:22.11E}'.format(Z) for Z in self.ccdata.atomnos - self.ccdata.coreelectrons]
+        else:
+            return ['{:22.11E}'.format(Z) for Z in self.ccdata.atomnos]            
+
+    def _nuclear_coords(self):
+        """Nuclear coordinates in Bohr."""
+        coord_template = '%22.11E %.12E %.12E'
+        to_bohr = lambda x: utils.convertor(x, 'Angstrom', 'bohr')
+        nuc_coords = [coord_template % tuple(to_bohr(coord))
+                      for coord in self.ccdata.atomcoords[-1]]
+        return nuc_coords
+
+    def _net_charge(self):
+        """Net charge on molecule."""
+        return '%22.11E' % self.ccdata.charge
+
+    def _no_electrons(self):
+        """Number of electrons in molecule."""
+        homos = self.ccdata.homos
+        charge = self.ccdata.charge if hasattr(self.ccdata, 'charge') else 0.0
+        if len(homos) > 1:
+            return int(sum(homos)) + 2
+        return int((homos[0] + 1)*2 - (charge % 2))
+
+    def _no_alpha_electrons(self):
+        """Number of alpha electrons in molecule."""
+        return int(numpy.ceil(self._no_electrons() / 2.0))
+
+    def _no_beta_electrons(self):
+        """Number of beta electrons in molecule."""
+        return self._no_electrons() - self._no_alpha_electrons()
+
+    def _spin_mult(self):
+        """Electronic Spin Multiplicity"""
+        return self.ccdata.mult
+
+    def _prim_centers(self):
+        """List of nuclear numbers upon which the primitive basis functions
+        are centered."""
+        prim_centers = []
+        for nuc_num, atom in enumerate(self.ccdata.gbasis, start=1):
+            for prims in atom:
+                prim_centers += [nuc_num] * num_orb[prims[0]]\
+                                * len(prims[1])
+
+        return _list_format(prim_centers, 10, '%d ')
+
+    def _rearrange_modata(self, data):
+        """Rearrange MO related data."""
+        prim_types = self._prim_types()
+        if isinstance(data, numpy.ndarray):
+            data = data.tolist()
+
+        pos_yyx = [key for key, val in enumerate(prim_types)
+                   if val == 17]
+        pos_yyz = [key for key, val in enumerate(prim_types)
+                   if val == 16]
+
+        if pos_yyx:
+            for pos in pos_yyx:
+                data.insert(pos-3, data.pop(pos))
```

Comment:
Consider not mutating arguments, or stating clearly in the docstring that this is the case.

Reply:
Docstring updated to explain the need of rearranging.

Reply:
Docstring updated to explain the need of such rearrangements.